### PR TITLE
Fix #3829 blicon ml readme

### LINKED ISSUE #3829: blicon: generate README in exported ml script zip
It's not obvious what to do after expanding the zip file, or what will happen at each stage.  Put that info into a README

### PR DESCRIPTION
**Pull request by mkeilman**
Adds a dynamically-build README file with instructions and explanations for specific data generation settings.

Also makes a few changes to the bash script:

- removes files from the scan directory if they exist
- lets the user choose to continue on and run SRW. If they opt not to, prints the instructions to do so later

### CODE REVIEW

File: sirepo/template/srw.py
Comment (robnagler) [inline sirepo/template/srw.py:392]: That's interesting. I didn't know that. I thought you always had to backslash "." in square brackets...

File: sirepo/template/srw.py
Comment (robnagler) [inline sirepo/template/srw.py:1904]: This is a lot of concepts in one line. I think this would be easier to read:

```py
for p in params:
    if p in e:
        for i, x in enumerate(e[p].fieldNames):
```

File: sirepo/template/srw.py
Comment (robnagler) [inline sirepo/template/srw.py:1907]: This is strange. Is there only one? The output will be a long string separated by underscores. No separator?

File: sirepo/package_data/static/js/srw.js
Comment (robnagler) [inline sirepo/package_data/static/js/srw.js:3261]: indentation.

File: sirepo/package_data/template/srw/README.txt.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/README.txt.jinja:14]: This line is too long, break it up so the "python ..." is on its own line.

File: sirepo/package_data/template/srw/README.txt.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/README.txt.jinja:13]: This line is too long, too.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/rsoptExport.sh.jinja:2]: It's better to do this if this is your intention:

```bash
for f in '{{ pyFileName }}' run_parallel_python.py; do
    rm -f "{{ runDir }}/$f"
done
```    

The output "Cleaning run directory {{ runDir }}..." is not useful. It's already happened so too late for the user. If you want to ask the user, then by all means prompt. Otherwise, they just lost data, and you don't have any way to recover.

The `-f` is important your code, too, because those files may or may not exist. You only know that the directory is not empty. That could be anything, the `rm` will output a message if the file doesn't exist, but not if `-f`.

The iteration is necessary only because there would be too many braces with:

`rm -f '{{ runDir }}/{{{ pyFileName }},run_parallel_python.py}`

which is the form I would like to use, and would use it if pyFileName was a shell variable, for example.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/rsoptExport.sh.jinja:7]: Quotes unnecessary. I would say running rsopt or better yet tell them the command that's being run.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/rsoptExport.sh.jinja:12]: This is cool. I didn't know about `$SECONDS`. However, I wouldn't bother printing it. It's an unnecessary output. There's a prompt below which is all that's needed.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (robnagler) [inline sirepo/package_data/template/srw/rsoptExport.sh.jinja:14]: Why do we want to prompt the user here? They could just control-C. If they are running this with `&`, they will have to answer a prompt. Was this a user request?

**Review by robnagler — CHANGES_REQUESTED** [review]
I gave some detailed script comments.

**Comment by mkeilman** [comment]
> I gave some detailed script comments.

Thank you - most of this is "first result on stackexchange"

**Comment by robnagler** [comment]
Glad you looked. I learned something.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:13]: bash does not seem to like the single quotes:
```
/home/vagrant/.pyenv/versions/py3/bin/python: can't open file ''NSLS-II-TES-beamline-rsOptExport.py'': [Errno 2] No such file or directory
```

Comment (robnagler) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:13]: Does pyFileName have quotes in it?

Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:13]: I'm investigating - it works without the single quotes but partially works using the ```eval``` syntax

Comment (robnagler) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:13]: This works for me (assuming x.py exists):

```sh
echo 'print("x")' > x.py
x="python 'x.py'"
eval "$x"
```

Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: a1227e2

The file name that comes out of the ```tail``` has single quotes around it. I think libensemble does that (via ```tools.save_libE_output```). The cut leaves a leading space (```' H_sample...'```) which did not matter to the backtick syntax I guess? Anyway I like regex better. Best would be to get the filename directly but both rsopt and libensemble take part in building it and I don't wanna.

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 3321937

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 3321937

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 3321937

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 3321937

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 3321937

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:22]: Not a user request, just thought an explicit opt-out would be polite. This also allows printing the instructions on "N". Not married to it

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 5421ff1

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 5421ff1

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 5421ff1

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:2]: dea7ee1

File: sirepo/package_data/template/srw/README.txt.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/README.txt.jinja]: 41a07e1

File: sirepo/package_data/template/srw/README.txt.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/README.txt.jinja]: 41a07e1

File: sirepo/template/srw.py
Comment (mkeilman) [reply sirepo/template/srw.py:1932]: Yes. This is the format used by the initial rsopt example (so e.g. ```beam_intensities_Aperture_size_horizontalSize_Aperture_size_verticalSize_10_runs.npy```). I suppose the idea is to have info about the run apparent at a glance but I can see it running into filename length limits in even moderately complex cases. May be worth getting input from @cchall...

File: sirepo/template/srw.py
Comment (mkeilman) [reply sirepo/template/srw.py]: fb40f8e

File: sirepo/package_data/static/js/srw.js
Comment (mkeilman) [reply sirepo/package_data/static/js/srw.js]: 57e72b3

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja:21]: Or, for another hot take, if they know to use ```&``` then they can use ```yes ```  :)

**Comment by mkeilman** [comment]
Ready for another look. Still a couple of open questions

File: sirepo/template/srw.py
Comment (cchall) [reply sirepo/template/srw.py]: I'm not sure I know what the file being named here is supposed to be. Is this the libensemble history file generated at the end of an rsopt run?

Comment (mkeilman) [reply sirepo/template/srw.py]: It's the SRW output, so it's nothing directly to do with rsopt. I think it came from an example of parallel execution - or maybe I invented it and forgot. Anyway it seems there is no reason for this format so I will just replace it with a static filename

Comment (mkeilman) [reply sirepo/template/srw.py]: (it came from the ```SRW_multiprocess.py``` example in ```rslight```):

```
    ####################### updates where input/output will be saved based on which offsets 
    save_str = ''
    if offsets_mirror1:
        
        save_str += 'mirror1_offsets_'
        
    if offsets_mirror2:
        
        save_str += 'mirror2_offsets_'
...
    ###### actual input and output files used for ML 
    np.save('datasets/beam_intensities_' + save_str + str(n_runs) + 'runs.npy', beams_all)
```

Comment (mkeilman) [reply sirepo/template/srw.py]: d4bb4d1

File: sirepo/package_data/template/srw/rsoptExport.sh.jinja
Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: Adding that [this fix](https://github.com/radiasoft/rsopt/issues/84) will make all the ```tail``` business moot

Comment (mkeilman) [reply sirepo/package_data/template/srw/rsoptExport.sh.jinja]: 8909343

Note that as of this writing, the ```output_file``` option lives in the devlop branch of rsopt. A merge here should wait until that change gets merged

**Comment by mkeilman** [comment]
ping for a re-look

**Pull request merged**
